lib/scout_apm/auto_instrument/rails.rb

```diff
@@ -0,0 +1,130 @@
+
+module ScoutApm
+  module AutoInstrument
+    module Rails
+      # It's possible that this code is invoked before Rails is loaded.
+      def self.controller_root
+        @controller_root ||= if defined? ::Rails
+          if defined? ::Rails.root
+            if !::Rails.root.nil?
+              File.join(::Rails.root, 'app', 'controllers')
+            end
+          end
+        end
+      end
+
+      # A general pattern to match Rails controller files:
+      CONTROLLER_FILE = /.*_controller.rb$/
+
+      # Whether the given path is likely to be a Rails controller.
+      # If `::Rails.root` is not defined, this will always return false.
+      def self.controller_path? path
+        if root = self.controller_root
+          path.start_with?(root) and path =~ CONTROLLER_FILE
+        end
+      end
+
+      def self.rewrite(path, code = nil)
+        code ||= File.read(path)
+
+        ast = Parser::CurrentRuby.parse(code)
+
+        # pp ast
+
+        buffer = Parser::Source::Buffer.new(path)
+        buffer.source = code
+
+        rewriter = Rewriter.new
+
+        # Rewrite the AST, returns a String with the new form.
+        rewriter.rewrite(buffer, ast)
+      end
+
+      class Rewriter < ::Parser::TreeRewriter
+        def initialize
+          super
+
+          # Keeps track of the parent - child relationship between nodes:
+          @nesting = []
+
+          # The stack of method nodes (type :def):
+          @method = []
+        end
+
+        def instrument(method_name, line, column)
+          ["::ScoutApm::AutoInstrument(\"\#{self.class}\\\##{method_name}:L#{line}:C#{column}\"){", "}"]
```

Comment:
The `desc` field of a layer may make sense to put an exact line & column number in. 

So the name can be `Foo/length` and desc is `line 3, column 8` or whatever format you want.

Reply:
The latest version of this puts the class and controller method with line number into the name field and the first line of code/expression into the description field. How do you feel about that? Can you try it and see if it makes sense?